### PR TITLE
Fix/mesh deploy hang/#4313

### LINKED ISSUE #4313: [mesheryctl] `mesh deploy` stops at verifying prerequisites
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

#### Current Behavior
Output showing current behavior:

```
➜  ~ mesheryctl mesh deploy
Verifying prerequisites...

➜  ~ 
```

Nothing happens beyond this message.

#### Desired Behavior
The command should tell the user that the `-a` is missing as a required argument.

#### Environment

- Meshery Client Version: stable-v0.5.62

---

#### Contributor Resources
- [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by Zanda256**
**Description**

This PR fixes #4313 

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Zanda256** [comment]
@piyushsingariya , please look into this. It stops the behaviour of hanging at Verifying pre-requisites but seems to shadow code that would prompt the user to choose an adapter.

File: mesheryctl/internal/cli/root/mesh/deploy.go [hunk 1]
```diff
@@ -16,12 +17,22 @@ import (
 	"github.com/spf13/viper"
 )
 
+// checkArgs checks whether the user has supplied an adapter(-a) argument
+func checkArgs(n int) cobra.PositionalArgs {
+	return func(cmd *cobra.Command, args []string) error {
+		if len(args) < n || args[0] == "" {
```
Comment (hexxdump) [inline mesheryctl/internal/cli/root/mesh/deploy.go:23]: should the check be for min arguments of length 2, i.e "-a" and "adapter-name" ?

Comment (Zanda256) [reply mesheryctl/internal/cli/root/mesh/deploy.go:23]: "-a" and "--adapter-name" are the same thing, so one of them has to be supplied. The command hangs at "Verifying pre-requisites" no arguments are supplied. But works with other valid arguments that are not "-a". So the number of arguments cannot be zero and the error message is intended to tell the user which argument is required.

Comment (hexxdump) [reply mesheryctl/internal/cli/root/mesh/deploy.go:23]: @Zanda256 I meant to ask, if user provides "-a" option and no valid adapter-name for e.g. meshery-linkerd, this checkArgs function would not alert the user, right ?
`mesheryctl mesh deploy -a`

**Review by hexxdump — APPROVED** [review]
LGTM

**Pull request merged**